modules/backendworker/backendworker.go

```diff
@@ -0,0 +1,254 @@
+
+	s.Service = services.NewBasicService(s.starting, s.running, s.stopping)
+	return s, nil
+}
+
+func (w *BackendWorker) starting(ctx context.Context) error {
+	if w.cfg.Poll {
+		w.store.EnablePolling(ctx, w)
+	}
+
+	return nil
+}
+
+func (w *BackendWorker) running(ctx context.Context) error {
+	// TODO: consider dropping the ticker and rely only on the backoff.
+	ticker := time.NewTicker(w.cfg.Interval)
+	defer ticker.Stop()
+
+	level.Info(log.Logger).Log("msg", "backend scheduler running")
+
+	b := backoff.New(ctx, w.cfg.Backoff)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-ticker.C:
+			if err := w.processCompactionJobs(ctx); err != nil {
+				level.Error(log.Logger).Log("msg", "error processing compaction jobs", "err", err, "backoff", b.NextDelay())
+				b.Wait()
+				continue
+			}
+
+			b.Reset()
+		}
+	}
+}
+
+func (w *BackendWorker) processCompactionJobs(ctx context.Context) error {
+	// TODO: the org ID is not used by the backend scheduler, but it is required
+	// by the request.  Figure out how to disable this requirement.
+
+	// Request next job
+	resp, err := w.backendScheduler.Next(user.InjectOrgID(ctx, w.workerID), &tempopb.NextJobRequest{
+		WorkerId: w.workerID,
+		Type:     tempopb.JobType_JOB_TYPE_COMPACTION,
+	})
+	if err != nil {
+		return fmt.Errorf("error getting next job: %w", err)
+	}
+
+	if resp == nil || resp.JobId == "" {
+		return fmt.Errorf("no jobs available")
+	}
+
+	if resp.Detail.Tenant == "" {
+		return w.failJob(ctx, resp.JobId, "received job with empty tenant")
+	}
+
+	level.Debug(log.Logger).Log("msg", "received job", "job_id", resp.JobId, "tenant", resp.Detail.Tenant)
+
+	blockMetas := w.store.BlockMetas(resp.Detail.Tenant)
+
+	// Collect the metas which match the IDs in the job
+	var sourceMetas []*backend.BlockMeta
+	for _, blockMeta := range blockMetas {
+		for _, blockID := range resp.Detail.Compaction.Input {
+			if blockMeta.BlockID.String() == blockID {
+				sourceMetas = append(sourceMetas, blockMeta)
+			}
+		}
+	}
+
+	// Execute compaction using existing logic
+	// err = w.store.Compact(ctx, sourceMetas, resp.Detail.Tenant)
+	err = w.compact(ctx, sourceMetas, resp.Detail.Tenant)
+	if err != nil {
+		return w.failJob(ctx, resp.JobId, fmt.Sprintf("error compacting blocks: %v", err))
+	}
+
+	// Mark job as complete
+	_, err = w.backendScheduler.UpdateJob(user.InjectOrgID(ctx, w.workerID), &tempopb.UpdateJobStatusRequest{
+		JobId:  resp.JobId,
+		Status: tempopb.JobStatus_JOB_STATUS_SUCCEEDED,
+	})
+	if err != nil {
+		return fmt.Errorf("failed marking job %q as complete: %w", resp.JobId, err)
+	}
+
+	return nil
+}
+
+func (w *BackendWorker) stopping(_ error) error {
+	return nil
+}
+
+func (w *BackendWorker) failJob(ctx context.Context, jobID string, errMsg string) error {
+	level.Error(log.Logger).Log("msg", "job failed", "job_id", jobID, "error", errMsg)
+
+	_, err := w.backendScheduler.UpdateJob(user.InjectOrgID(ctx, w.workerID), &tempopb.UpdateJobStatusRequest{
```

Comment:
i would make a wrapper func that you use for all uses of UpdateJob and Next that handles backoff. Something like:

```
func callSchedulerWithBackoff(ctx, func() err {
  w.backendScheduler.UpdateJob()
})
```

that way the same back off settings apply to all calls to the scheduler and all interactions between the worker and scheduler tolerate small outages in the same way.

it would use a loop like this one from mimir:  
https://github.com/grafana/mimir/blob/769906aa8f9d789759e783d972f0454238639e1f/pkg/usagestats/seed.go#L55